metric_learn/base_metric.py

```diff
@@ -410,6 +410,10 @@ def set_threshold(self, threshold):
     """
     check_is_fitted(self, 'preprocessor_')
 
+    if threshold is None or not isinstance(threshold, (int, float)):
```

Comment:
No need to check `is None` here, the isinstance check will handle None values as expected.

It may be even simpler to remove this checking and call `self.threshold_ = float(threshold)` instead. This will raise a ValueError if given invalid input, and also handles a broader range of float-like types.

Reply:
I agree with this simpler solution

Reply:
this also removes the need for a test I guess

Reply:
I agree as well. Perhaps writing a couple of test to make sure this raises the right error is still a good idea

Reply:
Ok! Will do